src/main/java/com/sourcegraph/cody/chat/CodeEditorFactory.java

```diff
@@ -139,14 +146,70 @@ public void mouseExited(@NotNull EditorMouseEvent event) {
             codeEditorButtons.setVisible(false);
           }
         };
-
     editor.addEditorMouseMotionListener(editorMouseMotionListener);
     editor.addEditorMouseListener(editorMouseListener);
     CodeEditorPart codeEditorPart = new CodeEditorPart(layeredEditorPane, editor);
     codeEditorPart.updateLanguage(language);
     return codeEditorPart;
   }
 
+  @Nullable private TransparentButton attributionIndicator() {
+    AttributionButton guardrailsButton = new AttributionButton("Attribution search running...");
+    guardrailsButton.setEnabled(false);
+    // TODO ACTUALLY GET ATTRIBUTION
+    CompletableFuture.runAsync(() -> {
```

Comment:
`runAsync` is sometimes problematic, because it uses current executor for spawning new thread, and it's not always clear what that will be as in IJ there are multiple various threads types/executors mixed together. Idiomatically we are trying to always use `ApplicationManager.getApplication().invokeLater` if all we need is just a background thread.

Reply:
I will leave an `invokeLater` and will send a follow-up PR with implementation of actually fetching attribution.